### PR TITLE
[@types/bcrypt] Fix type definitions

### PR DESCRIPTION
**Pull request by ghost**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: #68644
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

### CODE REVIEW

File: types/bcrypt/index.d.ts [hunk 1]
```diff
@@ -121,7 +121,7 @@ export declare function hash(
     data: string | Buffer,
     saltOrRounds: string | number,
     callback: (err: Error | undefined, encrypted: string) => any,
```
Comment (jakebailey) [inline types/bcrypt/index.d.ts:123]: Reading the implementation, it seems like it only returns the promise when it doesn't receive a callback, so this doesn't seem right to me.

Comment (MysteryBlokHed) [reply types/bcrypt/index.d.ts:123]: Agreed; this should probably be an overload

File: types/bcrypt/index.d.ts [hunk 2]
```diff
@@ -121,7 +121,7 @@ export declare function hash(
...
-): void;
+): Promise<string>;
```
Comment (MysteryBlokHed) [inline types/bcrypt/index.d.ts:124]:
```suggestion
): void;
export declare function hash(
    data: string | Buffer,
    saltOrRounds: string | number,
): Promise<string>;
```

Comment (MysteryBlokHed) [reply types/bcrypt/index.d.ts:124]: Oops, you’re right—I didn’t see that at the top. The return for this one should probably just be changed to `void`, then:

```suggestion
): void;
```

**Pull request closed without merging**